src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/HealthCheckFunctions.cs

```diff
@@ -1,63 +1,82 @@
 using GovUk.Education.ExploreEducationStatistics.Public.Data.Model.Database;
-using GovUk.Education.ExploreEducationStatistics.Public.Data.Services.Options;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Services.Interfaces;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
 using Microsoft.Azure.Functions.Worker;
-using Microsoft.Azure.Functions.Worker.Http;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.Options;
 
 namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Functions;
 
 // ReSharper disable once ClassNeverInstantiated.Global
 public class HealthCheckFunctions(
     ILogger<HealthCheckFunctions> logger,
     PublicDataDbContext publicDataDbContext,
-    IOptions<ParquetFilesOptions> parquetFileOptions)
+    IDataSetVersionPathResolver dataSetVersionPathResolver)
 {
-    [Function(nameof(CountDataSets))]
-    public async Task<string> CountDataSets(
+    [Function(nameof(HealthCheck))]
+    [Produces("application/json")]
+    public async Task<IActionResult> HealthCheck(
 #pragma warning disable IDE0060
-        [HttpTrigger(AuthorizationLevel.Function, "get")] HttpRequestData request)
+        [HttpTrigger(AuthorizationLevel.Anonymous, "get")] HttpRequest request)
 #pragma warning restore IDE0060
     {
+        var psqlConnectionHealthCheck = await CheckPsqlConnectionHealthy();
+        var fileShareMountHealthCheck = CheckFileShareMountHealth();
+        var healthCheckResponse = new HealthCheckResponse(
+            PsqlConnection: psqlConnectionHealthCheck,
+            FileShareMount: fileShareMountHealthCheck);
+
+        if (healthCheckResponse.Healthy)
+        {
+            return new OkObjectResult(healthCheckResponse);
+        }
+        
+        return new ObjectResult(healthCheckResponse){
+            StatusCode = StatusCodes.Status500InternalServerError
+        };
+    }
+    
+    private async Task<HealthCheckSummary> CheckPsqlConnectionHealthy()
+    {
+        logger.LogInformation("Attempting to test PSQL health");
+
         try
         {
-            var message = $"Found {await publicDataDbContext.DataSets.CountAsync()} datasets.";
-            logger.LogInformation(message);
-            return message;
+            await publicDataDbContext.DataSets.AnyAsync();
```

Comment:
I doubt there's any difference in effort on the database but if we wanted to make this check super generic and not depend on any particular table we could change this to `await publicDataDbContext.Database.ExecuteSqlRawAsync("SELECT 1");`

Reply:
I suppose this gives us the extra advantage that we can also test that the managed identity has correct privileges to access the "public" schema of the DB this way, so I'd probably be keen to keep it this way.

Reply:
That's fine by me, good point.